### PR TITLE
[6846] Update API documentation

### PR DESCRIPTION
**Pull request by tomtrentham**
### Context

https://trello.com/c/D4NO76Kg/6846-update-api-documentation

We want to have the API documented to make it easy for vendors and providers to use. This PR aims to make the documentation better and bring it up to date with the endpoints that have been developed so far.

### Changes proposed in this pull request

* Add information to existing /api-docs pages
* Add a help page
* Update reference page for v0.1 with endpoints
* Tweak the format and styling of the reference pages

### Guidance to review

* Do the pages provide enough information for vendors/providers to effectively use the API?
* Is the information clear? How could it be made clearer?

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: app/assets/stylesheets/api-docs.scss [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  padding: 3px 5px;
+  font-family: Monaco, Consolas, "Lucida Console", monospace;
+  font-size: 15px;
+  color: $govuk-text-colour;
+  word-break: break-word;
+  word-wrap: break-word;
+  overflow-wrap: break-word;
+  white-space: break-spaces;
+  background: govuk-colour("light-grey");
+}
```
Comment (defong) [inline app/assets/stylesheets/api-docs.scss:18]: Line ending please

File: app/views/api_docs/pages/home.md [hunk 2]
```diff
@@ -135,27 +134,30 @@ The number after the decimal point indicates a minor version. This is
...
+if version `1.2` is released then this URL will give you version `1.2` instead
 
 # Testing
 
-To get familiar with our system and perform testing, you can use our
-[sandbox environment](https://sandbox.register-trainee-teachers.service.gov.uk/).
+To get familiar with our system and perform testing, you can use
+[our sandbox environment](https://sandbox.register-trainee-teachers.service.gov.uk/).
+
+You will need an [authentication token](http://localhost:3000/api-docs/reference#authentication)
```
Comment (defong) [inline app/views/api_docs/pages/home.md:162]: Should it use localhost here?

**Review by defong — APPROVED** [review]

File: app/assets/stylesheets/api-docs.scss (hunk 1 above)
Comment (tomtrentham) [reply app/assets/stylesheets/api-docs.scss]: Fixed in https://github.com/DFE-Digital/register-trainee-teachers/pull/4133/commits/f2fb052d35e4f7c2da03676ba4b39a7fa9b3c3d7

File: app/views/api_docs/pages/home.md (hunk 2 above)
Comment (tomtrentham) [reply app/views/api_docs/pages/home.md]: No, it shouldn't. Fixed in https://github.com/DFE-Digital/register-trainee-teachers/pull/4133/commits/f2fb052d35e4f7c2da03676ba4b39a7fa9b3c3d7

**Pull request merged**